### PR TITLE
records migration: return records from new database

### PR DESCRIPTION
**Pull request by TBonnin**
I won't merge this PR today, monitoring the fetch and compare logs for now but I will appreciate a review.
It is behind a feature flag anyway.

## Describe your changes
Returns records from the new database if flag is enabled (`new-records-return:global`)

### CODE REVIEW

File: packages/server/lib/controllers/onboarding.controller.ts [hunk 1]
```diff
@@ -161,12 +162,31 @@ class OnboardingController {
                 payload.progress = 3;
             }
 
-            const getRecords = await syncDataService.getAllDataRecords(connectionId, DEMO_GITHUB_CONFIG_KEY, environment.id, DEMO_MODEL);
-            if (!getRecords.success) {
+            let getRecords: Result<CustomerFacingDataRecord[], Error> = resultErr('failed_to_get_records');
+            const shouldReturnNewRecords = await featureFlags.isEnabled('new-records-return', 'global', false);
+            if (shouldReturnNewRecords) {
+                const { error, response: connection } = await connectionService.getConnection(connectionId, DEMO_GITHUB_CONFIG_KEY, environment.id);
```
Comment (bodinsamuel) [inline packages/server/lib/controllers/onboarding.controller.ts]: Connection is already checked in `checkIfConnectionExists` it returns the id

File: packages/server/lib/controllers/onboarding.controller.ts [hunk 2]
```diff
@@ -161,12 +162,31 @@ class OnboardingController {
                 payload.progress = 3;
             }
 
-            const getRecords = await syncDataService.getAllDataRecords(connectionId, DEMO_GITHUB_CONFIG_KEY, environment.id, DEMO_MODEL);
-            if (!getRecords.success) {
+            let getRecords: Result<CustomerFacingDataRecord[], Error> = resultErr('failed_to_get_records');
```
Comment (bodinsamuel) [inline packages/server/lib/controllers/onboarding.controller.ts]: It's hiding the actual error (if any) but maybe it's alright during the transition?

File: packages/server/lib/controllers/onboarding.controller.ts [hunk 3]
```diff
@@ -26,10 +26,11 @@ import {
     createActivityLog,
```
Comment (bodinsamuel) [inline packages/server/lib/controllers/onboarding.controller.ts:26]: tbf you could migrate the onboarding now

**Review by bodinsamuel — COMMENTED** [review]
looks okay, but didn't tested locally 👌🏻

File: packages/server/lib/controllers/onboarding.controller.ts (hunk 2 above)
Comment (TBonnin) [reply packages/server/lib/controllers/onboarding.controller.ts]: we don't do anything with the actual error but yeah fixed in 9e577400

File: packages/server/lib/controllers/onboarding.controller.ts (hunk 1 above)
Comment (TBonnin) [reply packages/server/lib/controllers/onboarding.controller.ts]: fixed in 9e577400

**Review by bodinsamuel — APPROVED** [review]

**Pull request merged**